### ISSUE TITLE
[cmake] documentation cleanup (Trac #1654)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1646]
Get better at hiding the advanced stuff from what basic users need to know. Maybe make a subdirectory for all of that so it doesn't show up on the main index, but still exists if you look hard enough (or if we had functioning search).

For instance, this section is expert-level: http://software.icecube.wisc.edu/documentation/projects/cmake/pybindings.html#helpful-preprocessor-macros

I would also argue that it's time to hide all mentions of I3_PORTS in the basic documentation.

Migrated from https://code.icecube.wisc.edu/ticket/1654
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:24", 
    "description": "Get better at hiding the advanced stuff from what basic users need to know. Maybe make a subdirectory for all of that so it doesn't show up on the main index, but still exists if you look hard enough (or if we had functioning search).\n\nFor instance, this section is expert-level: http://software.icecube.wisc.edu/documentation/projects/cmake/pybindings.html#helpful-preprocessor-macros\n\nI would also argue that it's time to hide all mentions of I3_PORTS in the basic documentation.", 
    "reporter": "david.schultz", 
    "cc": "olivas", 
    "resolution": "fixed", 
    "_ts": "1550067204154158", 
    "component": "cmake", 
    "summary": "[cmake] documentation cleanup", 
    "priority": "minor", 
    "keywords": "documentation", 
    "time": "2016-04-25T16:48:27", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160425T17:07:00: 
* nega commented:

> If you've got ideas to improve documentation, go ahead and implement them


* nega changed status from "new" to "assigned"
* nega changed owner from "nega" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20160729T21:48:15: 
* david.schultz changed resolution from "" to "fixed"
* david.schultz changed status from "assigned" to "closed"
* david.schultz commented:

> In changeset 2642:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2642"
> separate cmake docs into an easier user guide and advanced sections. hide some mentions of I3_PORTS, referring to cvmfs or platforms instead. still more work to do, but this should be useful to the average user. closes #1646
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:23:51: nwhitehorn commented:

> In changeset 170310:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170310"
> separate cmake docs into an easier user guide and advanced sections. hide some mentions of I3_PORTS, referring to cvmfs or platforms instead. still more work to do, but this should be useful to the average user. closes #1646
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:24: 
* olivas changed milestone from "October 2016" to ""
* olivas commented:

> Milestone October 2016 deleted
